staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -301,16 +759,153 @@ devices:
 
 			expectResults: []any{allocatedTwoDeviceClaim},
 		},
+		"devices split across different slices": {
+			claimsToAllocate: objects(fourDeviceClaim),
+			classes:          objects(driverAClass),
+			slices:           objects(node1slice, node1slice2),
+			node:             node1,
+
+			expectResults: []any{allocatedFourDeviceClaim},
+		},
+		"obsolete slices": {
+			claimsToAllocate: objects(twoDeviceClaim),
```

Comment:
Can the tests be derived from the "simple" case as often as possible?

Here the relevant difference is the addition of `node1ObsoleteSlice`. That it allocates two devices seems unnecessary.

Reply:
You are probably right with saying that generating the test data with Go functions is better. Then instead of referencing pre-generated test data in all tests, each test can generate its own test data slightly differently and the difference between the tests will be more obvious.

Right now one has to remember all test data or jump back-and-forth to verify the correctness of each test case.

Reply:
In addition to that editing YAML is a pain in go code, at least with VSCode. It inserts tabs all the time and I had to replace them with spaces after the test runtime failures caused by yaml parser errors.